### ISSUE TITLE
0pdd.rb:58-60: Let's add GZIP compression to this...

### ISSUE THREAD

**Issue by 0pdd** [yegor256/0pdd#47]
The puzzle `41-ab139346` in `0pdd.rb` (lines 58-60) has to be resolved: Let's add GZIP compression to this output, since most XML files are rather big and it would be beneficial to see them compressed in the browser.

The [puzzle](http://www.yegor256.com/2009/03/04/pdd.html) was created by Yegor Bugayenko on 15-Feb-17. 

Estimate: 30 minutes, role: IMP.

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is removed from the source code.

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this issue

**Comment by 0crat** [comment]
Job [#47](https://github.com/yegor256/0pdd/issues/47) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Thanks for your contribution, @0pdd[/z](https://www.0crat.com/u/0pdd)! If you would be a member of [the project](https://www.0crat.com/p/C3T46CUJJ), you would now earn +15 reputation points, as explained in [§29](http://datum.zerocracy.com/pages/policy.html#29). You can join and apply to it, see [§2](http://datum.zerocracy.com/pages/policy.html#2).

**Comment by 0pdd** [comment]
@0pdd all puzzles are solved

**Comment by 0pdd** [comment]
@0pdd all puzzles are solved

**Comment by 0pdd** [comment]
@0pdd all puzzles are solved

**Comment by 0crat** [comment]
The job [#47](https://github.com/yegor256/0pdd/issues/47) assigned to @pbenety[/z](https://www.0crat.com/u/pbenety), here is [why](https://www.0crat.com/footprint/C3T46CUJJ/1451507251); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)com.zerocracy.Par@56bf3acd

**Comment by 0crat** [comment]
@pbenety[/z](https://www.0crat.com/u/pbenety) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @pbenety[/z](https://www.0crat.com/u/pbenety) resigned from [#47](https://github.com/yegor256/0pdd/issues/47), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -30 point(s) just awarded to @pbenety[/z](https://www.0crat.com/u/pbenety)

**Comment by pbenety** [comment]
@0crat wait

**Comment by 0crat** [comment]
The job [#47](https://github.com/yegor256/0pdd/issues/47) assigned to @vl3[/z](https://www.0crat.com/u/vl3), here is [why](https://www.0crat.com/footprint/C3T46CUJJ/1551532098); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0crat** [comment]
@vl3[/z](https://www.0crat.com/u/vl3) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @vl3[/z](https://www.0crat.com/u/vl3) resigned from [#47](https://github.com/yegor256/0pdd/issues/47), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -30 point(s) just awarded to @vl3[/z](https://www.0crat.com/u/vl3)

**Comment by 0crat** [comment]
The job [#47](https://github.com/yegor256/0pdd/issues/47) assigned to @golyalpha[/z](https://www.0crat.com/u/golyalpha), here is [why](https://www.0crat.com/footprint/C3T46CUJJ/1700854634); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by golyalpha** [comment]
@yegor256 Can you define "see them compressed in the browser"?
Do you mean, be able to download a zip file with the XML, or send the compressed XML to the browser in such a way that it deflates it before displaying it?

**Comment by 0crat** [comment]
@golyalpha[/z](https://www.0crat.com/u/golyalpha) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by golyalpha** [comment]
@yegor256 Still waiting on clarification.

**Comment by yegor256** [comment]
@golyalpha just add normal GZIP compression for HTTP responses.

**Comment by 0crat** [comment]
The user @golyalpha[/z](https://www.0crat.com/u/golyalpha) resigned from [#47](https://github.com/yegor256/0pdd/issues/47), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -30 point(s) just awarded to @golyalpha[/z](https://www.0crat.com/u/golyalpha)

**Comment by 0crat** [comment]
The job [#47](https://github.com/yegor256/0pdd/issues/47) assigned to @pdacostaporto[/z](https://www.0crat.com/u/pdacostaporto), here is [why](https://www.0crat.com/footprint/C3T46CUJJ/1870636469); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@pdacostaporto[/z](https://www.0crat.com/u/pdacostaporto) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by pdacostaporto** [comment]
@0crat waiting for PR review

**Comment by pdacostaporto** [comment]
@0crat waiting for review #230

**Comment by 0crat** [comment]
> @0crat waiting for review #230  ([here](https://github.com/yegor256/0pdd/issues/47#issuecomment-405128197))

@pdacostaporto Job [#47](https://github.com/yegor256/0pdd/issues/47) is already on hold

**Comment by pdacostaporto** [comment]
@0crat waiting for review #233

**Comment by 0crat** [comment]
> @0crat waiting for review #233 ([here](https://github.com/yegor256/0pdd/issues/47#issuecomment-405459212))

@pdacostaporto Job [#47](https://github.com/yegor256/0pdd/issues/47) is already on hold

**Comment by 0crat** [comment]
@dennis00010011b[/z](https://www.0crat.com/u/dennis00010011b) please review this job completed by @pdacostaporto[/z](https://www.0crat.com/u/pdacostaporto), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed
